test/jdk/java/text/Format/CompactNumberFormat/TestCompactPatternsValidity.java

```diff
@@ -101,16 +101,18 @@
             {COMPACT_PATTERN5, List.of(100, 1000, 30000), List.of("100", "1K", "K3")},
             {COMPACT_PATTERN6, List.of(20.99, 1000, 30000), List.of("21", ".1K", ".30K")},
             {COMPACT_PATTERN7, List.of(100, 1000, new BigInteger("12345678987654321")), List.of("100", "1K,", "12345678987654K,")},
-            {COMPACT_PATTERN8, List.of(new BigInteger("223565686837667632"), new BigDecimal("12322456774334.89766"), 30000, 3456.78),
-                    List.of("223566T", "12T", "30K", "3K")},
+            {COMPACT_PATTERN8, List.of(new BigInteger("223565686837667632"), new BigDecimal("12322456774334.89766"), 30000, 3456.78,
+                    new BigInteger("999999999999"), new BigDecimal("999999999999.0"), 999_999),
+                    List.of("223566T", "12T", "30K", "3K", "1T", "1T", "1M")},
```

Comment:
The case where 999_999_999 was formatted to 1000 million and now 1 billion may be added too.

Reply:
Sure, added the 1 bil test case.